### PR TITLE
Deprecate "Type" and support more devices for save_on_cpu

### PR DESCRIPTION
**Pull request by heidongxianhua**
Fixes #ISSUE_NUMBER
1、the class named "Type" has not been used anymore in anywhere, so I add warning message  to remove it in the future.
2、add a arg(default is "cuda") for save_on_cpu so that it can support more device type (like privateuse1)

### CODE REVIEW

**Comment by soulitzer** [comment]
Updated the title to reflect both things the PR is doing.

**Review by soulitzer — APPROVED** [review]
Thanks! Just a minor comment, otherwise looks good

File: torch/autograd/graph.py [hunk 1]
```diff
@@ -258,16 +258,16 @@ class save_on_cpu(saved_tensors_hooks):
...
     """
-    def __init__(self, pin_memory=False):
+    def __init__(self, pin_memory=False, device_type="cuda"):
         def pack_to_cpu(tensor):
             if not pin_memory:
                 return (tensor.device, tensor.cpu())
-
+            device_module = getattr(torch, device_type, torch.cuda)
             packed = torch.empty(
                 tensor.size(),
```
Comment (vadimkantorov) [inline torch/autograd/graph.py:267]: should this use `tensor.shape` instead? it's more modern, right?

**Pull request closed without merging**